src/net.h

```diff
@@ -1261,6 +1261,7 @@ struct NodeEvictionCandidate
     bool prefer_evict;
     bool m_is_local;
     Network m_network;
+    bool m_noban;
```

Comment:
At a high level, it's a little weird to add a flag to the eviction criteria that basically means "don't consider me for eviction". I assume this will prompt in future commits the reasonable question: should we bother adding nodes that can't be evicted to the eviction manager? (I don't know the answer to that question at this point, just pointing out that the discussion will come up later).

Reply:
It's obviously correct here btw, I didn't mean to imply otherwise. Just a little weird :)